### PR TITLE
[oracle] remove OCCI includes from ROOT headers

### PR DESCRIPTION
**Pull request by linev**
Seems to be, cling/llvm9 have problems to parse OCCI headers. 
With enabled oracle hsimple.C crashes making some message about accessing libOracle module
In any case such form is preferable for plugins

### CODE REVIEW

File: sql/oracle/CMakeLists.txt [hunk 1]
```diff
@@ -36,6 +36,6 @@ if(${GLIBCXX_USE_CXX11_ABI})
       COMPILE_DEFINITIONS _GLIBCXX_USE_CXX11_ABI=0)
 endif()
 
-target_compile_definitions(Oracle PUBLIC ${ORACLE_DEFINITIONS})
-target_include_directories(Oracle PUBLIC ${ORACLE_INCLUDE_DIR})
-target_link_libraries(Oracle PUBLIC ${ORACLE_LIBRARIES})
+target_compile_definitions(Oracle PRIVATE ${ORACLE_DEFINITIONS})
+target_include_directories(Oracle PRIVATE ${ORACLE_INCLUDE_DIR})
```
Comment (vgvassilev) [inline sql/oracle/CMakeLists.txt:40]:
```suggestion
target_include_directories(Oracle SYSTEM ${ORACLE_INCLUDE_DIR})
```

Comment (linev) [reply sql/oracle/CMakeLists.txt:40]: Why `SYSTEM`

Comment (linev) [reply sql/oracle/CMakeLists.txt:40]: These includes should not be exposed to any other tools

Comment (vgvassilev) [reply sql/oracle/CMakeLists.txt:40]: That'd append `-isystem` which suppresses warnings coming from these headers.

Comment (vgvassilev) [reply sql/oracle/CMakeLists.txt:40]: Ha, we can do `SYSTEM PRIVATE`

Comment (linev) [reply sql/oracle/CMakeLists.txt]: Ok, `SYSTEM PRIVATE` make sense then

**Review by Axel-Naumann — APPROVED** [review]
Wonderful, thanks!

**Comment by linev** [comment]
@Axel-Naumann, do we have on any Jenkins nodes OCCI installed?
Is it tested by chance while I only can prove on my single platform.

**Comment by linev** [comment]
On this PR build for all nodes `-Doracle=off` :(

**Comment by Axel-Naumann** [comment]
Can you open an issue, please, so we add one node with OCCI?

**Pull request merged**